greybox_generalize.py

```diff
@@ -0,0 +1,249 @@
+import numpy as np
+import pyomo.environ as pyo
+from scipy.sparse import coo_matrix
+from pyomo.contrib.pynumero.interfaces.external_grey_box import ExternalGreyBoxModel
+from pyomo.contrib.pynumero.interfaces.external_grey_box import ExternalGreyBoxBlock
+
+
+class LogDetModel(ExternalGreyBoxModel):
+    def __init__(
+        self,
+        n_parameters=2,
+        initial_fim=None,
+        use_exact_derivatives=True,
+        print_level=0,
+    ):
+        """
+        Greybox model to compute the log determinant of a sqaure symmetric matrix.
+
+        Arguments
+        ---------
+        n_parameters: int
+            Number of parameters in the model. The square symmetric matrix is of shape n_parameters*n_parameters
+        initial_fim: dict
```

Comment:
What are the required keys for the dictionary? Is the dictionary structure explained below?

Reply:
added